cmd/kubeadm/app/phases/apiconfig/setupmaster.go

```diff
@@ -49,6 +50,8 @@ func findMyself(client *clientset.Clientset) (*v1.Node, error) {
 }
 
 func attemptToUpdateMasterRoleLabelsAndTaints(client *clientset.Clientset) error {
+	validate.WaitForNode(client)
```

Comment:
The master components don't depend on CNI - or they didn't when I implemented self-hosted. They run on the host network.